src/igvfd/schemas/construct_library.json

```diff
@@ -0,0 +1,368 @@
+        {
+            "$ref": "mixins.json#/basic_item"
+        },
+        {
+            "$ref": "mixins.json#/accession"
+        },
+        {
+            "$ref": "mixins.json#/attribution"
+        },
+        {
+            "$ref": "mixins.json#/product_info"
+        },
+        {
+            "$ref": "mixins.json#/documents"
+        },
+        {
+            "$ref": "mixins.json#/references"
+        }
+    ],
+    "dependentSchemas": {
+        "crispr_library_details": {
+            "not": {
+                "required": [
+                    "mpra_library_details"
+                ]
+            },
+            "comment": "Specification of crispr_library_details is mutually exclusive with specification of mpra_library_details."
+        },
+        "mpra_library_details": {
+            "not": {
+                "required": [
+                    "crispr_library_details"
+                ]
+            },
+            "comment": "Specification of mpra_library_details is mutually exclusive with specification of crispr_library_details."
+        },
+        "scope": {
+            "oneOf": [
+                {
+                    "required": [
+                        "targeted_genes"
+                    ],
+                    "properties": {
+                        "scope": {
+                            "enum": [
+                                "genes"
+                            ]
+                        }
+                    }
+                },
+                {
+                    "required": [
+                        "targeted_loci"
+                    ],
+                    "properties": {
+                        "scope": {
+                            "enum": [
+                                "loci"
+                            ]
+                        }
+                    }
+                },
+                {
+                    "not": {
+                        "properties": {
+                            "scope": {
+                                "enum": [
+                                    "genes",
+                                    "loci"
+                                ]
+                            }
+                        }
+                    }
+                }
+            ]
+        },
+        "average_size": {
+            "not": {
+                "required": [
+                    "size_range"
+                ]
+            },
+            "comment": "Specification of average_size is mutually exclusive with specification of size_range."
+        },
+        "size_range": {
+            "not": {
+                "required": [
+                    "average_size"
+                ]
+            },
+            "comment": "Specification of size_range is mutually exclusive with specification of average_size."
+        }
+    },
+    "properties": {
+        "schema_version": {
+            "default": "1"
+        },
+        "accession": {
+            "accessionType": "FS"
```

Comment:
there is Jennifers' ticket that will change it to be DS instead of FS
probably need to wait for that one to get merged? or the other way around

Reply:
Rebased